src/components/sidebar-area/sidebar-area.tsx

```diff
@@ -83,14 +90,14 @@ export function SidebarArea() {
   }
 
   useEffect(() => {
-    if (data && data.length === 0) {
+    if (!isAgentsLoading && !data?.length && canCreateAgents) {
       handleCreateAgent()
     }
-  }, [data])
+  }, [data, isAgentsLoading, canCreateAgents])
 
   const hostname = useMemo(() => {
-    if (process.env.NEXT_PUBLIC_LETTA_SERVER_URL) {
-      const lettaServerHostname = new URL(process.env.NEXT_PUBLIC_LETTA_SERVER_URL).hostname
+    if (runtimeInfo?.LETTA_SERVER_URL) {
```

Comment:
Since runtime info is not immediately returned, we should add a loading resoponse ui

Reply:
Yup, you could also write as

```
['localhost', '127.0.0.1', '0.0.0.0'].includes(lettaServerHostname)  ? 'LOCAL SERVER'
: 'REMOTE SERVER'
```